plugin/storage/es/factory.go

```diff
@@ -218,7 +218,11 @@ func (f *Factory) CreateArchiveSpanReader() (spanstore.Reader, error) {
 	if !f.archiveConfig.Enabled {
 		return nil, nil
 	}
-	sr, err := createSpanReader(f.getArchiveClient, f.archiveConfig, true, f.logger, f.tracer)
+	suffix := "archive"
+	if f.archiveConfig.UseReadWriteAliases {
+		suffix += "-read"
+	}
+	sr, err := createSpanReader(f.getArchiveClient, f.archiveConfig, f.logger, f.tracer, suffix, true)
```

Comment:
@yurishkuro Not ideal to have to do this but this is roughly what we need to maintain backwards compatibility.

Reply:
I'm not sure if we want to expose a configuration option to override the default `read` and `write` alias but that could be one way to streamline this.

Reply:
* who adds read/write suffix to the index name?
* how does that work for non-archive, does it not use `-read` suffix?

Reply:
> who adds read/write suffix to the index name?

Its added by the reader and the writer but we need to handle the special case where the archive storage also has use_aliases set to true 

> how does that work for non-archive, does it not use -read suffix?

Yeah it does - its done by the reader and the writer. I made some changes to make it an else-if change. The current behavior is that if a suffix is passed in, we use that. If a suffix isn't passed in but use_aliases is set, then we use the default read/write aliases.

Reply:
that's why I don't follow - if reader/writer automatically add `-read/-write` suffix, why do you need to add it manually above?

Reply:
I add it above just to address the case where `use_aliases` wasn't initially set for the archive storage. Since we hardcode it to true for the archive storage now, we'll always set the index to be `archive-read` and `archive-write` in the reader/writer. But that would break backwards compatibility because previously it would just be `archive` for both the read and write alias.